vmm/src/lib.rs

```diff
@@ -2031,28 +2171,40 @@ impl RequestHandler for Vmm {
 
                     if !started {
                         warn!("Migration not started yet");
-                        Response::error().write_to(&mut socket)?;
+                        Response::error().write_to(&mut socket).map_err(|e| {
+                            MigratableError::MigrateReceive(anyhow!(
+                                "Error sending response: {}",
+                                e
+                            ))
+                        })?;
                         continue;
                     }
 
-                    let mut buf = [0u8; 4];
-                    let (_, file) = socket.recv_with_fd(&mut buf).map_err(|e| {
-                        MigratableError::MigrateReceive(anyhow!(
-                            "Error receiving slot from socket: {}",
-                            e
-                        ))
-                    })?;
+                    if let SocketStream::Unix(unix_socket) = &mut socket {
+                        let mut buf = [0u8; 4];
+                        let (_, file) = unix_socket.recv_with_fd(&mut buf).map_err(|e| {
+                            MigratableError::MigrateReceive(anyhow!(
+                                "Error receiving slot from socket: {}",
+                                e
+                            ))
+                        })?;
+
+                        if existing_memory_files.is_none() {
+                            existing_memory_files = Some(HashMap::new());
```

Comment:
> Ditto. Don't see why this change is needed.
> 
> Although HashMap::new() works here too, HashMap::default() is more flexible in general, e.g. support generic types that are determined at compile time.

I agree with your suggestion and it will not be modified here.